x/stake/handler.go

```diff
@@ -17,18 +16,26 @@ func NewHandler(k keeper.Keeper, govKeeper gov.Keeper) sdk.Handler {
 		// NOTE msg already has validate basic run
 		switch msg := msg.(type) {
 		case types.MsgCreateValidatorProposal:
+			if sdk.IsUpgrade(sdk.BEPHHH) {
+				return sdk.ErrMsgNotSupported("MsgCreateValidatorProposal disabled in BEP-HHH").Result()
+			}
 			return handleMsgCreateValidatorAfterProposal(ctx, msg, k, govKeeper)
 		case types.MsgRemoveValidator:
+			if sdk.IsUpgrade(sdk.BEPHHH) {
+				return sdk.ErrMsgNotSupported("MsgRemoveValidator disabled in BEP-HHH").Result()
+			}
 			return handleMsgRemoveValidatorAfterProposal(ctx, msg, k, govKeeper)
-		// disabled other msg handling
-		//case types.MsgEditValidator:
-		//	return handleMsgEditValidator(ctx, msg, k)
-		//case types.MsgDelegate:
-		//	return handleMsgDelegate(ctx, msg, k)
-		//case types.MsgBeginRedelegate:
-		//	return handleMsgBeginRedelegate(ctx, msg, k)
-		//case types.MsgBeginUnbonding:
-		//	return handleMsgBeginUnbonding(ctx, msg, k)
+		// Beacon Chain New Staking in BEP-HHH
+		case types.MsgCreateValidator:
+			return handleMsgCreateValidator(ctx, msg, k)
+		case types.MsgEditValidator:
+			return handleMsgEditValidator(ctx, msg, k)
+		case types.MsgDelegate:
```

Comment:
I think we need to enable `BEPHHH` check here, no check inside `handleMsgEditValidator`, and so on.

Reply:
As I can see from other BEP implementations, adding new message types will not check the BEP height.
I may need to register them with `upgrade.Mgr.RegisterMsgTypes`, like <https://github.com/bnb-chain/node/blob/master/app/app.go#L340>.